analyzer/consensus/queries.go

```diff
@@ -0,0 +1,308 @@
+package consensus
+
+import (
+	"fmt"
+)
+
+func makeLatestBlockQuery(chainID string) string {
```

Comment:
Errr.  So the only parameter for all of these routines is the chain ID, which also appears to be static from the config yeah?

```
type ChainID string

func NewChainID(cfg *config.Config) (ChainID, error) {
  // Blah blah blah
}

func (id ChainID) LatestBlockQuery() string { // This not being *sql.Stmt makes me sad.
  // Blah blah blah
}
```

Reply:
For the not being `*sql.Stmt` issue, maybe we could just pass all of these to [`pgx.ConnPool.Prepare`](https://pkg.go.dev/github.com/jackc/pgx#ConnPool.Prepare) on init. And then just call named statements everywhere.

Reply:
The pgx package claims to include a cache, so in theory there isn't a need to use Stmt.  Maybe something lke:
```
type QueryFactory {
  chainId string
  conn *db.Conn // Or whatever the type is supposed to be.
}

func (qf *QueryFactory) LastBlockQuery() *sql.Stmt {
   // Blah blah blah
}
```

Reply:
Alternatively, instead of fmt.Sprintf , maybe using `text/template` is cleaner.  Dunno.